From vapier@gentoo.org  Thu Jan 28 09:54:35 2010
From: Cliff Cai <cliff.cai@analog.com>
Date: Wed, 27 Jan 2010 20:40:03 -0500
Subject: USB: musb: set version of Blackfin version
To: linux-usb@vger.kernel.org, Felipe Balbi <felipe.balbi@nokia.com>
Cc: uclinux-dist-devel@blackfin.uclinux.org, Cliff Cai <cliff.cai@analog.com>
Message-ID: <1264642804-2093-1-git-send-email-vapier@gentoo.org>


From: Cliff Cai <cliff.cai@analog.com>

All current Blackfin parts are using RTL v1.9, but they don't expose the
hardware registers to probe this dynamically.  So hardcode the version to
v1.9 for now.

Signed-off-by: Cliff Cai <cliff.cai@analog.com>
Signed-off-by: Mike Frysinger <vapier@gentoo.org>
Cc: Felipe Balbi <felipe.balbi@nokia.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/musb/musb_regs.h |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/drivers/usb/musb/musb_regs.h
+++ b/drivers/usb/musb/musb_regs.h
@@ -533,7 +533,10 @@ static inline u8 musb_read_configdata(vo
 
 static inline u16 musb_read_hwvers(void __iomem *mbase)
 {
-	return 0;
+	/* This register is invisible on Blackfin, actually the MUSB
+	 * RTL version of Blackfin is 1.9, So just set it's value to 1.9.
+	 */
+	return 1 << 10 | 9;
 }
 
 static inline void __iomem *musb_read_target_reg_base(u8 i, void __iomem *mbase)
